localstack/utils/docker.py

```diff
@@ -596,5 +803,559 @@ def mountable_tmp_file():
         TMP_FILES.append(f)
         return f
 
+    @staticmethod
+    def append_without_latest(image_names):
+        suffix = ":latest"
+        for image in list(image_names):
+            if image.endswith(suffix):
+                image_names.append(image[: -len(suffix)])
+
+    @staticmethod
+    def tar_path(path, target_path, is_dir: bool):
+        f = tempfile.NamedTemporaryFile()
+        with tarfile.open(mode="w", fileobj=f) as t:
+            abs_path = os.path.abspath(path)
+            arcname = (
+                os.path.basename(path)
+                if is_dir
+                else (os.path.basename(target_path) or os.path.basename(path))
+            )
+            t.add(abs_path, arcname=arcname)
+
+        f.seek(0)
+        return f
+
+    @staticmethod
+    def untar_to_path(tardata, target_path):
+        target_path = Path(target_path)
+        with tempfile.NamedTemporaryFile() as dest:
+            for d in tardata:
+                dest.write(d)
+            dest.seek(0)
+            with tarfile.open(mode="r", fileobj=dest) as t:
+                if target_path.is_dir():
+                    t.extractall(path=target_path)
+                else:
+                    member = t.next()
```

Comment:
Could we potentially have multiple entries in the tar file (i.e., calling `t.next()` in a loop until `None` is returned)?

Reply:
My thought here was, that if the target is not a directory (so its a file or does not exist), multiple entries do not make any sense, since I cannot unpack them all to the same file anyway, but maybe I am overlooking something here?

Reply:
Ok, thanks for clarifying - makes sense. 👍